### ISSUE TITLE
Conditional execution

### ISSUE THREAD

**Issue by KrishnaPG** [brophdawg11/Automator.js#2]
This is good work. Thanks for sharing it.

I have a question: How to conditionally execute few steps in a chain/flow? 

For example,
1.    Ajax.get() some json
2.    if json contains some particular value, do additional step, such as `show login form and complete authentication`
3.    display the ajax results in brower

The 'show login and complete auth' is conditional based on the json results of first step. How to specify this kind of sequence?

**Comment by brophdawg11** [comment]
Take a look at the [Pass Through Values](https://github.com/brophdawg11/Automator.js#pass-through-values) section of the README, I think something like the following might work?  This is untested, but I can look at it in more detail when I have a bit more time.

```
function showLoginFormAndAuthenticate(ajaxData) {
    var dfd = $.Deferred();
    // Show the form
    // Perform authentication
    // Once authentication is complete, resolve 'dfd' with 'ajaxData' to pass it to the next step
    return dfd.promise();
}

new Automator().automate([
    function doAjax() {
        return $.get('...');
    },
    function showLoginForm(ajaxData) {
        if (ajaxData.shouldLogin) {
            return showLoginFormAndAuthenticate();
        }   
        return $.Deferred().resolve(ajaxData);
    },
    function displayResults(ajaxData) {
        // Display results from ajaxData
    }
]);
```
